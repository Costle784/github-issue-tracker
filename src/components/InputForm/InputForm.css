.form-container {
    margin: 10% auto;
    width: 30%;
    font-size: 2em;
    text-align: center;
}

form * {
    display: block;
}

.api-key-input {
    height: 2em;
    width: 50%;
    border-radius: 8px;
    padding: 3px;
    margin: 1em auto;
}

.submit-button {
    background-color: #eb7100;
    width: 12em;
    border-radius: 8px;
    border: None;
    padding: 8px;
    margin: 0 auto;
}

.submit-button:hover {
    cursor: pointer;
}
