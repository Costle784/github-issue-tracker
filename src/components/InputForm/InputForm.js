import React, { useState } from "react";
import { useDispatch } from "react-redux";
import { updateApiKey } from "../../actions/inputForm";
import { useHistory } from "react-router-dom";
import "./InputForm.css";

export default function InputForm() {
    const [key, updateKey] = useState("");
    const dispatch = useDispatch();
    const history = useHistory();

    const handleSubmit = (e) => {
        e.preventDefault();
        dispatch(updateApiKey(key));
        history.push(`/repos`);
    };

    return (
        <div className="form-container">
            <form onSubmit={handleSubmit}>
                <label>Input API Key: </label>
                <input
                    className="api-key-input"
                    value={key}
                    type="text"
                    onChange={(e) => updateKey(e.target.value)}
                />
                <button
                    type="submit"
                    className="submit-button"
                    disabled={!key || (key && key.trim().length === 0)}
                >
                    Submit
                </button>
            </form>
        </div>
    );
}
