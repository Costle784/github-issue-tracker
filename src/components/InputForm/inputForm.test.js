import React from "react";
import InputForm from "./InputForm";
import renderer from "react-test-renderer";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
    useSelector: jest.fn(),
    useDispatch: () => mockDispatch,
}));

describe("API key form snapshot test", () => {
    it("should render a form to input an API key", () => {
        const renderedForm = renderer.create(<InputForm />).toJSON();
        expect(renderedForm).toMatchSnapshot();
    });
});
