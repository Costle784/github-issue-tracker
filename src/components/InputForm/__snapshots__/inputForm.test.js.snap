// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API key form snapshot test should render a form to input an API key 1`] = `
<div
  className="form-container"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Input API Key: 
    </label>
    <input
      className="api-key-input"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="submit-button"
      disabled={true}
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
